core/src/main/java/org/apache/iceberg/ClientPoolImpl.java

```diff
@@ -36,6 +38,10 @@
   private volatile int currentSize;
   private boolean closed;
 
+  private static volatile AtomicInteger openCount = new AtomicInteger(0);
```

Comment:
I am not sure if I understand the purpose of these three variables

Reply:
This is just for counting the number of connections，maybe it can be used as a metric in the future.

Reply:
I'm a little confused on the variables as well.

If I'm not mistaken, it's the total number of connections ever opened and the total number of connections closed? Looking at the declaration, I would have thought that it was the current number of open connections, not the number of connections ever opened.

My only indication that it might be otherwise came from the unit test - https://github.com/apache/iceberg/pull/2649/files#diff-8b2035adec2ca6d247f624aeb6dbb2cc3789bd188541c500de4319b969813758R78-R80

```java
  private void waitingForCleanup() {
     long startTime = System.currentTimeMillis();
     while ((ClientPoolImpl.openCount().get() != ClientPoolImpl.closeCount().get()) 
```

Maybe you could add a small comment?

Reply:
Thanks for your time, yes your right, it's the total number of connections ever opened and the total number of connections closed，just for unit test, maybe the log is also useful.